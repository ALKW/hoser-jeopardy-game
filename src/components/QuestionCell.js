import React, { Component } from 'react';


class QuestionCell extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <td onClick={() => {this.props.openQuestion(this.props.category, this.props.value);} }>
        ${this.props.value}
      </td>
    );
  }
}

export default QuestionCell;
