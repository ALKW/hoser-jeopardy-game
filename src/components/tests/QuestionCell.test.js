import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";

import QuestionCell from "./../QuestionCell";

beforeAll(() => {
  configure({ adapter: new Adapter() });
});

describe("QuestionCell Component", () => {
  it("renders correctly when isAnswered is false", () => {
    const component = shallow(<QuestionCell />);
    expect(component).toMatchSnapshot();
  });
  it("renders empty td when isAnswered is true", () => {
    const component = shallow(<QuestionCell isAnswered />);
    expect(component).toMatchSnapshot();
  });

  it("calls openQuestion on click", () => {
    const mockOpenQuestion = jest.fn();
    const component = shallow(<QuestionCell openQuestion={mockOpenQuestion} />);

    component.find("td").simulate("click");
    expect(mockOpenQuestion).toHaveBeenCalled();
  });

  it("renders value prop", () => {
    const expectedValue = "What is ReactJS?";
    const component = shallow(<QuestionCell value={expectedValue} />);

    //Dollar sign is prepended to value string because it is a dollar amount
    expect(component.text()).toBe(`$${expectedValue}`);
  });
});
