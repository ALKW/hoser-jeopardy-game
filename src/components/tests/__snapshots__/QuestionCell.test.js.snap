// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionCell Component renders correctly when isAnswered is false 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "$",
        undefined,
      ],
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      "$",
      undefined,
    ],
    "type": "td",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "$",
          undefined,
        ],
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        "$",
        undefined,
      ],
      "type": "td",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;

exports[`QuestionCell Component renders empty td when isAnswered is true 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    isAnswered={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {},
    "ref": null,
    "rendered": null,
    "type": "td",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": "td",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
